docs/DeveloperGuide.md

```diff
@@ -376,6 +376,95 @@ when used, instead of an invalid or unexpected entry.
             when the system is running other process concurrently.
         * Handling of behaviour for specific date types in terms of comparing dates may require more work.
 
+### Interaction record feature
+
+#### Implementation
+
+Each `Person` object contains its own list of `ContactedInfo` objects. These objects are stored in a List in each `Person` object.
+The class diagram below shows how the recently contacted information feature is implemented in the Model component.
+
+<img src="images/ModelContactedInfo.png" width="300" />
+
+##### ContactedInfo
+
+A `ContactedInfo` object contains the information regarding recent interactions for a specific client.
+
+The sequence diagram below shows how the add recently contacted information feature is parsed.
+
+<img src="images/AddContactedInfoCommandParserSeqDiagram.png" width="650" />
+
+The class diagaram below shows how ContactedInfo is implemented.
+
+<img src="images/ContactedInfo.png" width="550" />
+
+`ContactedInfo` objects have the following characteristics:
+
+* A `ContactedInfo` object has two private data members (final), `recentDate` of `RecentDate` object representing that date of interaction,
+  and `description` of `Description` object, representing the description of the interaction. Both objects would distinguish one ContactedInfo object from another.
+* Any two `ContactedInfo` objects are not unique if both `description` and `recentDate` is equal.
+* `ContactedInfo` objects can be sorted, and the sorted order is the reverse ordering of the `RecentDate`.
+
+###### RecentDate
+
+`RecentDate` is an object that stores information regarding the interaction date for `ContactedInfo`. `RecentDate` object inherits from `DocumentedDate` object.
+The sequence diagram below shows the creation of a RecentDate object.
+
+<img src="images/RecentDateCreationSequenceDiagram.png" width="650" />
```

Comment:
Too big

Reply:
And I'm concerned there are too many similar diagrams, that's why I didn't add in the first place

Reply:
Maybe take out 410 to 412